runtime/server/auth/metricsview_policy.go

```diff
@@ -0,0 +1,114 @@
+package auth
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/golang-lru/simplelru"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime/compilers/rillv1"
+)
+
+var lru *simplelru.LRU
+
+var openPolicy = &ResolvedMetricsViewPolicy{
+	HasAccess: true,
+	Filter:    "",
+	Include:   nil,
+	Exclude:   nil,
+}
+
+type ResolvedMetricsViewPolicy struct {
+	HasAccess bool
+	Filter    string
+	Include   []string
+	Exclude   []string
+}
+
+func ResolveMetricsViewPolicy(userClaims *UserClaims, mv *runtimev1.MetricsView) (*ResolvedMetricsViewPolicy, error) {
+	if mv.Policy == nil {
+		return nil, nil
+	}
+
+	if lru == nil {
+		var err error
+		lru, err = simplelru.NewLRU(1000, nil)
+		if err != nil {
+			// or just log and move on with no cache?
+			panic(err)
+		}
+	}
+
+	// take hash of mv and userClaims
+	key := fmt.Sprintf("%v:%v:%v:%v:%v:%v", userClaims.Name, userClaims.Email, userClaims.Domain, strings.Join(userClaims.Groups, ","), mv.Name, mv.CatalogUpdatedOn)
+	cached, ok := lru.Get(key)
+	if ok {
+		return cached.(*ResolvedMetricsViewPolicy), nil
+	}
+
+	if userClaims.IsOpen {
+		return openPolicy, nil
+	}
```

Comment:
I think when aud in nil (local mode) and if attributes are set we apply policy otherwise we don't. So of the attribute values are set in the UI policies are applied otherwise it has the same behaviour as current, would be useful during development.